### PR TITLE
Adding optional tag and new parameters to GDAL miscellaneous algs #3287

### LINKED ISSUE #3287: [processing][needs-docs] Expose resampling and format options in the gdaladdo
Original commit: https://github.com/qgis/QGIS/commit/1233a21ac2fbe7931ef9ede09d59179dbf94ae77 by web-flow

[processing][needs-docs] expose resampling and format options in the gdaladdo (fix #20432)

### PR DESCRIPTION
**Pull request by rosaguilar**
### Description

Goal:

Ticket: fix #3287

- [ x] Backport to LTR documentation is required

### Minimal requirements for merging *(for maintainers)*

- [ ] The description of this PR is coherent with the manual and does not provide wrong information.
- [ ] This PR passes the checks.

### CODE REVIEW

File: source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst [hunk 1]
```diff
@@ -114,11 +120,31 @@ Parameters
...
-  Output raster file.
+``Virtual`` [raster]
+  Output raster file
+
+``Open output file after running the algorithm`` [boolean]
+  Add the output raster to the map canvas
+
+  Default: *True*
+
+  By clicking on ``Run as Batch Process..``, the algorithm can be executed as a batch process
```
[the remark this answers is not in the document]
Comment (rosaguilar) [reply source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]: I agree with you but I was not sure

File: source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst [hunk 2]
```diff
@@ -114,11 +120,31 @@ Parameters
...
   Default: *False*
 
+
+``Add alpha mask band to VRT when source raster has none`` [boolean]
+  Adds an alpha mask band to the VRT when the source raster have none
+
+``Override projection for the output file`` [enumeration]
+  Optional
+
+  A VRT will be created even if the input datasets have not the same projection.
```
[the remark this answers is not in the document]
Comment (rosaguilar) [reply source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]: I don't think so. When you click this option, in the console appears -allow_projection_difference

File: source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst [hunk 3]
```diff
@@ -114,11 +120,31 @@ Parameters
...
+
 Outputs
 .......
 
-``Output layer`` [raster]
-  Output raster file.
+``Virtual`` [raster]
+  Output raster file
+
+``Open output file after running the algorithm`` [boolean]
```
[the remark this answers is not in the document]
Comment (rosaguilar) [reply source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]: Good to know.

File: source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst [hunk 4]
```diff
@@ -169,12 +195,58 @@ Parameters
...
+
+  In this section multiple options can be passed to the output format driver
+   See `format specific documentation for legal creation options for each format
+   <https://www.gdal.org/formats_list.html>`_.
+
+   The button |signPlus| lets you add an option.
+   The button |signMinus| deletes an option previously added. By clicking on :guilabel:`validate`
+   you can verify whether the creation options are legal to the given output format.
+
+``Profile`` [enumeration]
```
[the remark this answers is not in the document]
Comment (rosaguilar) [reply source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]: I hesitate with this as well. Thanks!

**Comment by rosaguilar** [comment]
@DelazJ, thanks for all the revision and suggestions. I am sorry but  I am not sure if I should commit the suggestions, or will you do so?

File: source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (rosaguilar) [reply source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]: Sorry. My fault. You are right.

**Comment by DelazJ** [comment]
Afaics only the PR author can apply suggestions. You can commit if you agree with the change (if needed, see the tip in https://github.com/qgis/QGIS-Documentation/pull/3723/files for some "guidelines").
Actually, it's as you want; there's no obligation in using the "commit suggestions" button. You can simply toggle your file to edit as explained [here](https://docs.qgis.org/3.4/en/docs/documentation_guidelines/first_contribution.html#make-corrections) and type in the required changes (or do that on your local branch and push commit). As you want! Choose the most convenient process to you.
_Hope that I understood and answered clearly_

File: source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst [hunk 5]
```diff
@@ -169,13 +184,44 @@ Parameters
...
+
+    Options:
+
+    * 0 --- Default
+    * 1 --- No compression
+    * 2 --- Low compression
+    * 3 --- High compression
+    * 4 --- JPEG compression
+
+    Default: *1*
```
[the remark this answers is not in the document]
Comment (rosaguilar) [reply source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]: I skipped the empty value (it does not make sense to me enumerating an empty value as 0). So, the list starts at 1.

[the remark this answers is not in the document]
Comment (DelazJ) [reply source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]: I'll merge as is but I don't agree with your choice and actually I don't agree with the way we were treating this widget. I've tracked the profiles in source code and found that the empty entry is actually a placeholder for custom profiles you can create in Settings --> Options --> GDAL dialog (see #3755). And actually these profiles are not parameters of the algs, just some convenient GUI widgets that adds the actual parameters and these are the ones we should mention (if needed). 
We'll need to clean and harmonize the description of the "Additional Creation options" of all algs at some point.

[the remark this answers is not in the document]
Comment (rosaguilar) [reply source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]: I agree with you that it is better to clean and harmonize as much as possible, e.g., by linking to the recent description of the additional creation options. Feel free to update as you consider. Thanks for all the suggestions and reviews.

**Comment by DelazJ** [comment]
@rosaguilar the backport to 3.4 fails. Would you mind giving it a try, please? Instructions are provided in the error message above. Thanks.

**Comment by rosaguilar** [comment]
> @rosaguilar the backport to 3.4 fails. Would you mind giving it a try, please? Instructions are provided in the error message above. Thanks.

I will try. Thanks!

**Pull request merged**